File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.dart
class Solution {
  int maxWidthOfVerticalArea(List<List<int>> points) {
    points.sort((a,b) => a[0].compareTo(b[0]));
    int res = 0;
    for (int i = 1; i < points.length; i++) res = max(res,  points[i][0] - points[i - 1][0]);
    return res;
  }
}

File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.rs
impl Solution {
	pub fn max_width_of_vertical_area(mut points: Vec<Vec<i32>>) -> i32 {
		points.sort_unstable();
		points.windows(2).map(|w| w[1][0] - w[0][0]).max().unwrap_or(0)
	}
}